sql/src/main/java/org/apache/druid/sql/calcite/schema/SegmentMetadataCache.java

```diff
@@ -403,6 +403,13 @@ public void start() throws InterruptedException
   @VisibleForTesting
   void refresh(final Set<SegmentId> segmentsToRefresh, final Set<String> dataSourcesToRebuild) throws IOException
   {
+    long startMillis = System.currentTimeMillis();
+
+    emitter.emit(ServiceMetricEvent.builder().build(
```

Comment:
It would be better to emit the dimension `dataSource` with this metric. So we should probably emit it from `refreshSegmentsForDatasource` instead.

Reply:
Should we report both overall time and per datasource time?

Reply:
No, just add the dimension `dataSource` while emitting the metric. Aggregation is not required. It can be summed up wherever the metric is being collected.